sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -268,10 +268,11 @@ private void CloseOutputBuilder(Stream stream, OutputBuilder outputBuilder, bool
             using var sw = new StreamWriter(stream, defaultStreamWriterEncoding, DefaultStreamWriterBufferSize, leaveStreamOpen);
             sw.NewLine = "\n";
 
+            if (_config.HeaderText != string.Empty && !isMethodClass)
```

Comment:
Where will it be handled for the `methodClass` if not here?

Reply:
Also, won't https://github.com/microsoft/ClangSharp/issues/79 also apply to the single-file case if you have a header file without methods? So shouldn't https://github.com/microsoft/ClangSharp/blob/master/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs#L86 be fixed too?

Something like:
```cs
                using var sw = new StreamWriter(stream, defaultStreamWriterEncoding, DefaultStreamWriterBufferSize, leaveStreamOpen);
                {
                    sw.NewLine = "\n";
                    sw.Write(_config.HeaderText);

                    foreach (var usingDirective in usingDirectives)
                    {
                        sw.Write("using");
                        sw.Write(' ');
                        sw.Write(usingDirective);
                        sw.WriteLine(';');
                    }

                    sw.WriteLine();
                }
```

Reply:
From my tests this doesn't seem to happen, and the header always ends up getting written. I am not completely sure if I have tested every edge case though.

Maybe a better approach would be to change CloseOutputBuilder to accept an "additional usings" parameter and have it write everything, including the header?

Reply:
Even in the case where there are no using statements for the single file?

Reply:
Whats the output in that case?

Say if, for example, you just have a header file containing:
```cs
struct MyStruct
{
    int value;
}
```

Reply:
Ok, looks like the exact case above is fine, but if you do:
```
#include <stdint.h>

struct MyStruct
{
    intptr_t value;
};
```

Then you get the license file printed twice.

Reply:
```cs
// Copyright
using System;

// Copyright

namespace LLVMSharp
{
    public partial struct MyStruct
    {
        [NativeTypeName("intptr_t")]
        public IntPtr value;
    }
}
```

Reply:
This should probably be...
```cs
if (_config.GenerateMultipleFiles)
{
    if (_config.HeaderText != string.Empty)
    {
        sw.WriteLine(_config.HeaderText);
    }

    if (outputBuilder.UsingDirectives.Any()
    {
        ...
    }
}
```

Reply:
(The `!isMethodClass` was also causing issues, as the license wasn't generating for that file)

Reply:
:+1: